homeassistant/components/unifiprotect/views.py

```diff
@@ -0,0 +1,211 @@
+
+from .const import DOMAIN
+from .data import ProtectData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def async_generate_thumbnail_url(
+    event_id: str,
+    nvr_id: str,
+    width: int | None = None,
+    height: int | None = None,
+) -> str:
+    """Generate URL for event thumbnail."""
+
+    url_format = ThumbnailProxyView.url or "{nvr_id}/{event_id}"
+    url = url_format.format(nvr_id=nvr_id, event_id=event_id)
+
+    params = {}
+    if width is not None:
+        params["width"] = str(width)
+    if height is not None:
+        params["height"] = str(height)
+
+    return f"{url}?{urlencode(params)}"
+
+
+@callback
+def async_generate_event_video_url(event: Event) -> str:
+    """Generate URL for event video."""
+
+    _validate_event(event)
+    if event.start is None or event.end is None:
+        raise ValueError("Event is ongoing")
+
+    url_format = VideoProxyView.url or "{nvr_id}/{camera_id}/{start}/{end}"
+    url = url_format.format(
+        nvr_id=event.api.bootstrap.nvr.id,
+        camera_id=event.camera_id,
+        start=event.start.isoformat(),
+        end=event.end.isoformat(),
+    )
+
+    return url
+
+
+@callback
+def _client_error(message: Any, code: HTTPStatus) -> web.Response:
+    _LOGGER.warning("Client error (%s): %s", code.value, message)
+    if code == HTTPStatus.BAD_REQUEST:
+        return web.Response(body=message, status=code)
+    return web.Response(status=code)
+
+
+@callback
+def _400(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.BAD_REQUEST)
+
+
+@callback
+def _403(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.FORBIDDEN)
+
+
+@callback
+def _404(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.NOT_FOUND)
+
+
+@callback
+def _validate_event(event: Event) -> None:
+    if event.camera is None:
+        raise ValueError("Event does not have a camera")
+    if not event.camera.can_read_media(event.api.bootstrap.auth_user):
+        raise PermissionError(f"User cannot read media from camera: {event.camera.id}")
+
+
+class ProtectProxyView(HomeAssistantView):
+    """Base class to proxy request to UniFi Protect console."""
+
+    requires_auth = True
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize a thumbnail proxy view."""
+        self.hass = hass
+        self.data = hass.data[DOMAIN]
+
+    def _get_data_or_404(self, nvr_id: str) -> ProtectData | web.Response:
+        all_data: list[ProtectData] = []
+
+        for data in self.data.values():
+            if isinstance(data, ProtectData):
+                if data.api.bootstrap.nvr.id == nvr_id:
+                    return data
+                all_data.append(data)
+        return _404("Invalid NVR ID")
+
+
+class ThumbnailProxyView(ProtectProxyView):
+    """View to proxy event thumbnails from UniFi Protect."""
+
+    url = "/api/ufp/thumbnail/{nvr_id}/{event_id}"
```

Comment:
```suggestion
    url = "/api/unifiprotect/thumbnail/{nvr_id}/{event_id}"
```

Please match the namespace of the integration